src/RestApi/Utilities/ProductSummary.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Helper class to format a short summary of content for a product.
+ *
+ * @package WooCommerce/Blocks
+ */
+
+namespace Automattic\WooCommerce\Blocks\RestApi\Utilities;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Product Summary class.
+ */
+class ProductSummary {
+	/**
+	 * Original content from the product.
+	 *
+	 * @var string
+	 */
+	protected $content = '';
+
+	/**
+	 * Constructor.
+	 *
+	 * @param \WC_Product $product The product to generate a summary for.
+	 */
+	public function __construct( \WC_Product $product ) {
+		$this->content = $product->get_short_description();
+
+		if ( ! $this->content ) {
+			$this->content = $product->get_description();
+		}
+	}
+
+	/**
+	 * Return the formatted summary.
+	 *
+	 * @param int $max_words Word limit for summary.
+	 * @return string
+	 */
+	public function get_summary( $max_words = 25 ) {
+		$summary = $this->content;
+
+		if ( $max_words && $this->get_word_count( $summary ) > $max_words ) {
+			$summary = $this->generate_summary( $summary, $max_words );
+		}
+
+		return \wc_format_content( $summary );
+	}
+
+	/**
+	 * Get the word count.
+	 *
+	 * @param string $content HTML Content.
+	 * @return int Length
+	 */
+	protected function get_word_count( $content ) {
+		return \str_word_count( \wp_strip_all_tags( $content ) );
```

Comment:
The idea is that we only resort to trimming if the first paragraph is too long, to avoid cutting sentences in half. 

`wp_trim_words` isn't perfect for some languages though it states:

```
This function is localized. For languages that count ‘words’ by the individual character (such as East Asian languages), the $num_words argument will apply to the number of individual characters.
```

Ill add a test to cover this and make sure it works as expected.

Reply:
> The idea is that we only resort to trimming if the first paragraph is too long, to avoid cutting sentences in half.

Ahh, that makes sense. Good idea 👍 .

> I'll add a test to cover this and make sure it works as expected.
Thanks!